hack/.ci/prepare_release

```diff
@@ -25,34 +25,50 @@ apk add --no-cache \
     ca-certificates \
     make \
     bash \
-    go \
     git \
-    musl-dev \
     curl \
     openssl \
     tar \
     gzip \
-    gcc \
     sed
 
-GOLANG_VERSION="$(sed -rn 's/FROM (eu\.gcr\.io\/gardener-project\/3rd\/golang|golang):([^ ]+).*/\2/p' < "$repo_root_dir/Dockerfile")"
+# create virtual package with the dev tools
+echo "Installing dev tools"
+apk add --no-cache --virtual .build-deps \
+    gcc \
+    go \
+    musl-dev
 
-export \
-    GOROOT="$(go env GOROOT)" \
-    GOOS="$(go env GOOS)" \
-    GOARCH="$(go env GOARCH)" \
-    GOHOSTOS="$(go env GOHOSTOS)" \
-    GOHOSTARCH="$(go env GOHOSTARCH)"
+GOLANG_VERSION="$(sed -rn 's/FROM (eu\.gcr\.io\/gardener-project\/3rd\/golang|golang):([^ ]+).*/\2/p' < "$repo_root_dir/Dockerfile")"
 
-echo "Downloading go $GOLANG_VERSION"
+# alpine is using musl-libc instead of glibc, therefore we cannot use the avilable pre-built binaries from golang, but have to build them ourselves from source.
+# refs:
+#   - https://stackoverflow.com/a/45987284
+#   - https://github.com/docker-library/golang/blob/f300e60ca19c3b98cfcf01ca112af2ac10104320/1.16/alpine3.14/Dockerfile
+echo "Downloading go src $GOLANG_VERSION"
+rm -rf /usr/local/go
 wget -q -O - "https://golang.org/dl/go$GOLANG_VERSION.src.tar.gz" | tar zx -C /usr/local
+
+# Configure golang environment
+echo "Building and installing go $GOLANG_VERSION"
+export \
+  PATH="/usr/local/go/bin":$PATH \
+  GOARCH="$(go env GOARCH)" \
+  GOOS="$(go env GOOS)" \
+  GOROOT_BOOTSTRAP="$(go env GOROOT)"
+export GOHOSTOS="$GOOS" \
+  GOHOSTARCH="$GOARCH"
 cd /usr/local/go/src
 echo "Executing make on go $GOLANG_VERSION"
-./make.bash
+./make.bash -v
```

Comment:
I guess we can skip the verbose output

```suggestion
./make.bash
```

Reply:
Actually, I would prefer to keep it on as it is helpful when we need to debug something in the installation.

Reply:
What is shown in the verbose output that is useful? I only see some pkg names.

Reply:
You know, exactly this step was failing when the release image was based on alpine 3.14 and we have no clue what actually is failing. I do not want to spend again several days looking into this when I can more clearly see what failed.
Having more logs never hearts.